src/Serializer/AbstractItemNormalizer.php

```diff
@@ -488,4 +498,51 @@ protected function normalizeRelation(PropertyMetadata $propertyMetadata, $relate
 
         return $iri;
     }
+
+    /**
+     * Sets the object to populate if allowed to and not set already. If you pass identifiers, it will search for them,
+     * otherwise it falls back to the provided IdentifiersExtractor.
+     *
+     * @param mixed  $data
+     * @param string $class
+     * @param array  $context
+     */
+    protected function setObjectToPopulate($data, string $class, array &$context)
+    {
+        if (isset($context[self::OBJECT_TO_POPULATE]) || !\is_array($data)) {
+            return;
+        }
+
+        $updateAllowed = $context['api_allow_update'] ?? false;
+        $identifiers = $this->getIdentifiersForDenormalization($class);
+        $identifiersData = \array_intersect_key($data, array_flip($identifiers));
+
+        if (0 === \count($identifiersData)) {
+            return;
+        }
+
+        if (!$updateAllowed) {
+            throw new InvalidArgumentException('Update is not allowed for this operation.');
```

Comment:
Should not this exception be thrown before the previous condition (the `return;`)?

Reply:
don't think so, we only throw if identifiers are provided.